flake.nix

```diff
@@ -0,0 +1,129 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+  };
+
+  outputs = inputs:
+    inputs.flake-parts.lib.mkFlake { inherit inputs; } {
+      # TODO(XXX): in theory this flake could support aarch64-linux,
+      #   x86_64-darwin and aarch64-darwin, but it is untested.
+      systems = [ "x86_64-linux" ];
+      perSystem = { config, self', pkgs, lib, system, ... }:
+        let
+          buildToolsVersion = "30.0.3";
+          platformVersion = "33";
+          rustTargets = [ "x86_64-linux-android" ];
```

Comment:
With fresh eyes it seems silly to not derive this from the system` arg given to the `perSystem` func by `mkFlake`. Will try to fix.

Reply:
Fixed.